internal/server/healthService.go

```diff
@@ -103,5 +103,47 @@ func (h *HealthService) GetMentalHealthScoreForUser(
 	return res, err
 }
 
+func (h *HealthService) DeleteHealthDataForUser(
+	ctx context.Context,
+	req *pbhealth.DeleteHealthDataForUserRequest,
+) (*pbhealth.DeleteHealthDataForUserResponse, error) {
+	var err error
+	var numDeleted uint32
+
+	switch x := req.Data.(type) {
+	case *pbhealth.DeleteHealthDataForUserRequest_All:
+		numDeleted, err = h.db.DeleteMentalHealthLogs(ctx, req.UserID, nil, x.All)
+		break
+	case *pbhealth.DeleteHealthDataForUserRequest_DateToRemove:
+		numDeleted, err = h.db.DeleteMentalHealthLogs(ctx, req.UserID, x.DateToRemove, true)
```

Comment:
This looks like a bug to me, you are saying "true" to all, but this is meant to be the case where we delete a specific day.

Reply:
Good catch!